moodle-scalable-cluster-ubuntu/azuredeploy.json

```diff
@@ -30,58 +30,96 @@
+            "metadata": {
+                "description": "Switch to install Moodle Object FS plugins (with Azure Blob storage)"
+            },
+            "type": "bool"
+        },
         "installO365pluginsSwitch": {
             "defaultValue": false,
             "metadata": {
                 "description": "Switch to install Moodle Office 365 plugins"
             },
             "type": "bool"
         },
+        "installGdprPluginsSwitch": {
+            "defaultValue": false,
+            "metadata": {
+                "description": "Switch to install Moodle GDPR plugins. Note these require Moodle versions 3.4.2+ or 3.3.5+ and these will be included by default in Moodle 3.5"
+            },
+            "type": "bool"
+        },
         "installElasticSearchSwitch": {
             "defaultValue": false,
             "metadata": {
                 "description": "Switch to install Moodle ElasticSearch plugins & VMs"
             },
             "type": "bool"
         },
-        "storageAccountType": {
-            "defaultValue": "Standard_LRS",
-            "allowedValues": [
-                "Standard_LRS",
-                "Standard_GRS",
-                "Standard_ZRS"
-            ],
+        "htmlLocalCopySwitch": {
+            "defaultValue": true,
             "metadata": {
-                "description": "Storage Account type"
+                "description": "Switch to create a local copy of /moodle/html or not"
+            },
+            "type": "bool"
+        },
+        "siteURL": {
+            "defaultValue": "www.example.org",
+            "metadata": {
+                "description": "URL for Moodle site"
             },
             "type": "string"
         },
-        "dbServerType": {
-            "defaultValue": "mysql",
+        "moodleVersion": {
             "allowedValues": [
-                "mysql",
-                "postgres"
+                "MOODLE_34_STABLE",
+                "v3.4.2",
+                "v3.4.1",
+                "MOODLE_33_STABLE",
+                "MOODLE_32_STABLE",
+                "MOODLE_31_STABLE",
+                "MOODLE_30_STABLE",
+                "MOODLE_29_STABLE"
             ],
+            "defaultValue": "MOODLE_34_STABLE",
             "metadata": {
-                "description": "Database type"
+                "description": "The Moodle version you want to install."
             },
             "type": "string"
         },
-        "fileServerType": {
-            "defaultValue": "gluster",
-            "allowedValues": [
-                "gluster",
-                "nfs"
-            ],
+        "sshPublicKey": {
             "metadata": {
-                "description": "File server type: GlusterFS, Azure Files (CIFS)--disabled due to too slow perf, NFS--not highly available"
+                "description": "ssh public key"
+            },
+            "type": "string"
+        },
+        "sshUsername": {
+            "defaultValue": "azureadmin",
```

Comment:
remove this - increases surface area

Reply:
We took the approach of providing as many reasonable default values as possible to minimize the number of unfilled required params, allowing beginning customers to try out the templates without having to figure out all the parameters. Once they get confidence and ready to deploy for their production, they can customize all these parameters (just like the siteURL parameter as above). I think we'd like to keep it this way, as it has always been. @rgardler -- Would you agree?

Reply:
Right approach, but not with something that compromises security... having a default username makes it easier to attack the app

Reply:
Understood

Reply:
this still needs to be removed